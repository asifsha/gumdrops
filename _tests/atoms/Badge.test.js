/* globals mount */
import React from 'react';
import Badge from '../../components/atoms/Badge';

describe('Expect <Badge>', () => {
    it('to render', () => {
        const props = {
            text: 'This is text',
            context: 'inverse',
            className: 'foo-class',
            style: {
                width: '1000px'
            }
        };

        const wrapper = mount(<Badge {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to render properly with empty', () => {
        const props = {
            text: 'Ignored text',
            context: 'inverse',
            className: 'foo-class',
            empty: true,
            style: {
                width: '1000px'
            }
        };

        const wrapper = mount(<Badge {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
