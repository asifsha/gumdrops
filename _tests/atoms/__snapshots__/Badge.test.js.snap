// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Badge> to render 1`] = `
<Badge
  className="foo-class"
  context="inverse"
  empty={false}
  style={
    Object {
      "width": "1000px",
    }
  }
  text="This is text"
>
  <span
    className="gds-badge foo-class gds-badge--inverse"
    style={
      Object {
        "width": "1000px",
      }
    }
  >
    This is text
  </span>
</Badge>
`;

exports[`Expect <Badge> to render properly with empty 1`] = `
<Badge
  className="foo-class"
  context="inverse"
  empty={true}
  style={
    Object {
      "width": "1000px",
    }
  }
  text="Ignored text"
>
  <span
    className="gds-badge foo-class gds-badge--inverse gds-badge--empty"
    style={
      Object {
        "width": "1000px",
      }
    }
  />
</Badge>
`;
